src/sources/aws_s3/mod.rs

```diff
@@ -0,0 +1,251 @@
+use super::util::MultilineConfig;
+use crate::{
+    config::{DataType, GlobalOptions, SourceConfig, SourceDescription},
+    dns::Resolver,
+    line_agg,
+    rusoto::{self, RegionOrEndpoint},
+    shutdown::ShutdownSignal,
+    Pipeline,
+};
+use futures::future::{FutureExt, TryFutureExt};
+use rusoto_core::Region;
+use rusoto_s3::S3Client;
+use rusoto_sqs::SqsClient;
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use std::convert::TryInto;
+
+pub mod sqs;
+
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Compression {
```

Comment:
It might be nice to pull this up and share it with the `file` source in the future.